cmake/ccache.cmake

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+if(MSVC AND WITH_CCACHE)
+  if(CMAKE_VERSION VERSION_LESS 3.24)
+    # ccache >= 4.8 requires compile batching turned off that is available since CMake 3.24.
+    # See https://github.com/ccache/ccache/wiki/MS-Visual-Studio
+    set(WITH_CCACHE "OFF, CMake >= 3.24 is required")
+  else()
+    set(ccache_hints)
+    if(EXISTS "$ENV{ChocolateyInstall}")
+      # Bypass a shim executable provided by Chocolatey.
```

Comment:
Is this not something that users of this package manger should be fixing on their own machine? In any case, the comment doesn't explain why it's needed (same goes for all other comments here).

Reply:
> Is this not something that users of this package manger should be fixing on their own machine?

That's indeed unfortunate that Chocalately's `ccache.exe` shim executable can't be moved or copied without being broken. But it is our intention to copy it in order to integrate with MSVC, therefore, I believe that this is our responsibility to provide a workaround for such a case. Besides, it is only a single path hint.

Of course, this workaround can be removed in the future if it leads to extra maintenance burden.

> In any case, the comment doesn't explain why it's needed (same goes for all other comments here).

I’ve reworked the comments in the hope of addressing your comment.

Reply:
> But it is our intention to copy it in order to integrate with MSVC.

I still don't understand why we even need to do this though, to make ccache "work".

Reply:
[Reworked](https://github.com/hebasto/bitcoin/pull/215#issuecomment-2173378806).